src/rules/index.js

```diff
@@ -77,6 +78,7 @@ export default {
   "media-feature-value-dollar-variable": mediaFeatureValueDollarVariable,
   "no-dollar-variables": noDollarVariables,
   "no-duplicate-dollar-variables": noDuplicateDollarVariables,
+  "no-non-numeric-dimensions": noNonNumericDimensions,
```

Comment:
I have been thinking a bit about the name. It should probably start with the "category", e.g. [number-max-precision](https://github.com/stylelint/stylelint/blob/master/lib/rules/number-max-precision/README.md), [length-zero-no-unit](https://github.com/stylelint/stylelint/blob/master/lib/rules/length-zero-no-unit/README.md) as examples from stylelint.

How about `dimension-no-non-numeric-values` or something similar @rambleraptor ?

Reply:
Example of similar naming from stylelint: [shorthand-property-no-redundant-values](https://github.com/stylelint/stylelint/blob/master/lib/rules/shorthand-property-no-redundant-values/README.md)